toplevel/opttoploop.ml

```diff
@@ -641,22 +651,29 @@ let loop ppf =
   run_hooks After_setup;
   load_ocamlinit ppf;
   while true do
+    let log = Location.init_log ppf in
     let snap = Btype.snapshot () in
     try
       Lexing.flush_input lb;
       Location.reset();
       first_line := true;
       let phr = try !parse_toplevel_phrase lb with Exit -> raise PPerror in
-      let phr = preprocess_phrase ppf phr  in
+      let phr = preprocess_phrase log phr  in
       Env.reset_cache_toplevel ();
       if !Clflags.dump_parsetree then Printast.top_phrase ppf phr;
       if !Clflags.dump_source then Pprintast.top_phrase ppf phr;
-      ignore(execute_phrase true ppf phr)
+      ignore(execute_phrase true log phr);
+      Misc.Log.flush log
     with
     | End_of_file -> raise (Compenv.Exit_with_status 0)
-    | Sys.Break -> fprintf ppf "Interrupted.@."; Btype.backtrack snap
+    | Sys.Break ->
+        Misc.Log.logf "status" log "Interrupted.@.";
+        Btype.backtrack snap
```

Comment:
there is no flush here, or on `Enf_of_file` or `PPerror`, I don't know whether this is intentional.

Reply:
(maybe a `with_log` function would sort those things out)